include/swift/AST/CASTBridging.h

```diff
@@ -126,178 +170,151 @@ extern "C" {
 ///
 /// \returns a diagnostic instance that can be extended with additional
 /// information and then must be finished via \c SwiftDiagnostic_finish.
-BridgedDiagnostic SwiftDiagnostic_create(void *diagnosticEngine,
-                                         BridgedDiagnosticSeverity severity,
-                                         const void *_Nullable sourceLoc,
-                                         const unsigned char *_Nullable text,
-                                         long textLen);
+BridgedDiagnostic Diagnostic_create(BridgedDiagnosticEngine cDiags,
+                                    BridgedDiagnosticSeverity severity,
+                                    BridgedSourceLoc cLoc,
+                                    BridgedString cText);
 
 /// Highlight a source range as part of the diagnostic.
-void SwiftDiagnostic_highlight(
-    BridgedDiagnostic diag, const void *_Nullable startLoc, const void *_Nullable endLoc);
+void Diagnostic_highlight(
+    BridgedDiagnostic cDiag, BridgedSourceLoc cStartLoc, BridgedSourceLoc cEndLoc);
 
 /// Add a Fix-It to replace a source range as part of the diagnostic.
-void SwiftDiagnostic_fixItReplace(BridgedDiagnostic diag,
-                                  const void *_Nullable replaceStartLoc,
-                                  const void *_Nullable replaceEndLoc,
-                                  const unsigned char *_Nullable newText,
-                                  long newTextLen);
+void Diagnostic_fixItReplace(BridgedDiagnostic cDiag,
+                             BridgedSourceLoc cStartLoc,
+                             BridgedSourceLoc cEndLoc,
+                             BridgedString cReplaceText);
 
 /// Finish the given diagnostic and emit it.
-void SwiftDiagnostic_finish(BridgedDiagnostic diag);
+void Diagnostic_finish(BridgedDiagnostic cDiag);
 
 BridgedIdentifier
-SwiftASTContext_getIdentifier(void *ctx, const unsigned char *_Nullable str,
-                              long len);
+ASTContext_getIdentifier(BridgedASTContext cContext, BridgedString cStr);
 
-void *SwiftImportDecl_create(void *, void *, void *, char, void *,
-                             BridgedArrayRef, BridgedArrayRef);
+void *ImportDecl_create(BridgedASTContext cContext, BridgedDeclContext cDeclContext, BridgedSourceLoc cImportLoc, char, BridgedSourceLoc cKindLoc,
+                        BridgedArrayRef path, BridgedArrayRef cPathLocs);
 
-void *SwiftTopLevelCodeDecl_createStmt(void *ctx, void *DC, void *startLoc,
-                                       void *element, void *endLoc);
-void *SwiftTopLevelCodeDecl_createExpr(void *ctx, void *DC, void *startLoc,
-                                       void *element, void *endLoc);
+void *TopLevelCodeDecl_createStmt(BridgedASTContext cContext, BridgedDeclContext cDeclContext, BridgedSourceLoc cStartLoc,
+                                  void *element, BridgedSourceLoc cEndLoc);
```

Comment:
Out of curiosity: What is `void *element`? Are you intending to write a wrapper type for that in a follow-up PR?

Reply:
The underlying statement. And... maybe :P